sql/core.go

```diff
@@ -252,6 +252,8 @@ type Lockable interface {
 // ConvertToBool converts a value to a boolean. nil is considered false.
 func ConvertToBool(ctx *Context, v interface{}) (bool, error) {
 	switch b := v.(type) {
+	case []uint8:
+		return false, nil
```

Comment:
Should we look at the value in the byte array and return true if it's not 0? 

Looks like MySQL treats binary data as true if it's non-zero:
```
set @var1 = cast(1 as binary);
set @var2 = cast(0 as binary);

 select if(@var1, "true", "false");
+----------------------------+
| if(@var1, "true", "false") |
+----------------------------+
| true                       |
+----------------------------+

 select if (@var2, "true", "false");
+-----------------------------+
| if (@var2, "true", "false") |
+-----------------------------+
| false                       |
+-----------------------------+
```

Reply:
I initally set it to return false because of this test case:
```sql
mysql> select if(char('1'), 'true', 'false');
+--------------------------------+
| if(char('1'), 'true', 'false') |
+--------------------------------+
| false                          |
+--------------------------------+
1 row in set, 1 warning (0.0079 sec)
```

Turns out it's just doing a string conversion first under the hood.